src/logged-evaluation.nix

```diff
@@ -132,15 +132,19 @@ let
 
             export IN_NIX_SHELL=impure
 
-            # https://github.com/NixOS/nix/blob/92d08c02c84be34ec0df56ed718526c382845d1a/src/nix-build/nix-build.cc#
+            # https://github.com/NixOS/nix/blob/2.3.1/src/nix-build/nix-build.cc#L418
```

Comment:
I have a fairly weak preference towards using actual commits instead of tags -- and optionally adding a `(tag: 2.3.1)` at the end. In this URL it is a bit unclear if 2.3.1 is a branch or a tag itself, with a bit of added worry about tags being technically mutable. Same goes for the other URLs.

Reply:
Thanks - I'll address this once we've figured out the more high level issues with this PR.